cpp/src/arrow/builder.h

```diff
@@ -315,7 +327,11 @@ inline uint8_t expanded_uint_size(uint64_t val, uint8_t current_int_size) {
 
 class ARROW_EXPORT AdaptiveUIntBuilder : public AdaptiveIntBuilderBase {
  public:
+#ifdef ARROW_NO_DEFAULT_MEMORY_POOL
   explicit AdaptiveUIntBuilder(MemoryPool* pool);
```

Comment:
Fixing. I may need to twiddle the Doxyfile a little bit so it expands the macro properly